### PR TITLE
Clean up naming in image builder workflow

### PR DESCRIPTION
**Pull request by szepeviktor**
First I've started up with adding checks for Dockerfiles using [Hadolint](https://github.com/szepeviktor/byte-level-care/blob/master/.github/workflows/docker.yml).
Then I've read the current `docker.yml` file and got mental dizziness 😵

This PR cleans up naming and a few problems in the workflow file.
Replaces ❌esoteric names with publicly understandable ones.
One of my sources: https://docs.docker.com/engine/reference/commandline/build/

### CODE REVIEW

**Comment by szepeviktor** [comment]
How should I test it? I have no permission to start this workflow manually.

File: .github/workflows/images.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: Docker builds
+name: Images
```
[the remark this answers is not in the document]
Comment (szepeviktor) [reply .github/workflows/images.yml:1]: That was my trigger!
The Dockerfile "linting" workflow is called [docker.yml](https://github.com/szepeviktor/byte-level-care/blob/master/.github/workflows/docker.yml) ❕
This workflow is not about Docker but about Docker Images.

Reverting the name change ... 🏃🏻‍♂️

**Review by tillkruss — APPROVED** [review]

**Comment by szepeviktor** [comment]
I think the naming is esoteric again.

**Comment by szepeviktor** [comment]
The output looks like this project is not open-souce but for internal use only.

**Pull request merged**